### PR TITLE
Fix start script and improve monit script

### PR DESCRIPTION
**Pull request by SuchiSmita-N**
Due to a known bug with bosh (see issue for more information https://github.com/bosh-elastic-stack/elasticsearch-boshrelease/issues/25 ), we need to change the start script to start Elasticsearch process in a subshell.

But with this change we also need to change the monit script in order to
call ES healthcheck, because that's now the proper way for monit to report ES as running.

As far as we know, regarding monit, each cycle has a duration of 45
seconds. By looping for 10 cycles, we give to ES a maximum of 450seconds
until the process is successfully running. If the healthcheck doesn't
return a 200 within those 10 cycles (450 sec), then monit will stop the
process and report it as failing.

### CODE REVIEW

**Comment by making** [comment]
Thanks for sending this PR!
I'm aware that some users are enabling x-pack security feature, do you think this change will work when security is enabled?

**Comment by SuchiSmita-N** [comment]
@making Why do you think it would, I ask because we don't have X-pack enabled so we don't get to test.  I did have a look on the docs, Looks like It shouldn't.  But I am not aware of the internal setup of X-pack .

**Comment by making** [comment]
I just made sure that you have already tested.
OK, I'll check if the health check endpoint is open to everyone even when security is enabled.

**Comment by making** [comment]
It looks like anonymous access can be supported.

https://discuss.elastic.co/t/anonymous-user-to-access-cluster-health/150269/2

**Comment by SuchiSmita-N** [comment]
Nice, Thanks for checking that. Do we need to change anything at our end or it's alright ?

**Comment by making** [comment]
it looks good to me. I'll merge this PR after confirming an anonymous access by myself.

File: jobs/elasticsearch/templates/bin/ctl [hunk 1]
```diff
@@ -38,11 +38,13 @@ case $1 in
...
     sysctl -q -w vm.max_map_count=262144
 
+    (
     exec chpst -u $RUNAS:$RUNAS \
        /var/vcap/packages/elasticsearch/bin/elasticsearch \
-      -p $PIDFILE \
       <%= p("elasticsearch.exec.options", []).join(' ') %> \
       >>$LOG_DIR/elasticsearch.stdout.log 2>>$LOG_DIR/elasticsearch.stderr.log
+    ) &
+    echo $! > $PIDFILE
```
Comment (making) [inline jobs/elasticsearch/templates/bin/ctl:47]: Is this change related to this issue?

**Comment by making** [comment]
Merged via ec68050f074d4eb9cea41fe7916d21e52e30bfb9 .

I've made this feature optional and disabled by default so that it won't break existing deployments.
You can toggle it by `monit.enable_http_healthcheck` property.

File: jobs/elasticsearch/templates/bin/ctl (hunk 1 above)
Comment (antoniocascais) [reply jobs/elasticsearch/templates/bin/ctl:47]: Yes, it is: by executing the process in a subshell we make sure that the PID file exists once monit looks up for it. Otherwise we could end up in the situation where monit did not find the PID file and report the process as `execution failed`, but the process would still be running anyway.

Thanks for making it optional, it's a good idea :)

**Comment by SuchiSmita-N** [comment]
Thanks :)
